### PR TITLE
feat: group creation for non-fully-connected-graphs [WPB-2294]

### PR DESCRIPTION
**Pull request by tlebon**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-2294" title="WPB-2294" target="_blank">StoryWPB-2294</a>  [Web] Conversation creation errors on non-fully connected graph
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

properly handles errors for non-fully-connected-graph croup creation

### Solutions

Error Handling, modals.

### CODE REVIEW

File: src/i18n/en-US.json [hunk 1]
```diff
@@ -571,6 +571,10 @@
   "groupCreationPreferencesErrorNameShort": "At least 1 character",
   "groupCreationPreferencesHeader": "Create group",
   "groupCreationPreferencesPlaceholder": "Group name",
+  "groupCreationPreferencesNonFederatingHeadline": "Group can’t be created",
+  "groupCreationPreferencesNonFederatingEditList": "Edit Participants List",
+  "groupCreationPreferencesNonFederatingLeave": "DiscardGroupCreation",
```
Comment (V-Gira) [inline src/i18n/en-US.json]:
```suggestion
  "groupCreationPreferencesNonFederatingLeave": "Discard Group Creation",
```

File: src/script/components/Modals/GroupCreation/GroupCreationModal.tsx [hunk 2]
```diff
@@ -227,6 +233,44 @@ const GroupCreationModal: React.FC<GroupCreationModalProps> = ({
...
         }
       } catch (error) {
+        const NON_FEDERATING_BACKENDS = 409;
+
+        if (isAxiosError<NonFederatingBackendsData>(error) && error.response?.status === NON_FEDERATING_BACKENDS) {
+          const tempName = groupName;
+          setIsShown(false);
+          const backendString = error.response?.data!.non_federating_backends?.join(', and ');
+          const replaceBackends = replaceLink(
+            'https://support.wire.com/hc/en-us/articles/9357718008093-Backend',
```
Comment (V-Gira) [inline src/script/components/Modals/GroupCreation/GroupCreationModal.tsx]:
```suggestion
            'https://support.wire.com/hc/articles/9357718008093',
```

removing lang and title allow support for other languages

File: src/script/components/Modals/GroupCreation/GroupCreationModal.tsx [hunk 3]
```diff
@@ -227,6 +233,44 @@ const GroupCreationModal: React.FC<GroupCreationModalProps> = ({
           createNavigate(generateConversationUrl(conversation.qualifiedId))(event);
         }
       } catch (error) {
+        const NON_FEDERATING_BACKENDS = 409;
```
Comment (przemvs) [inline src/script/components/Modals/GroupCreation/GroupCreationModal.tsx]: Move it above component :)

**Review by przemvs — APPROVED** [review]

File: src/script/components/Modals/GroupCreation/GroupCreationModal.tsx (hunk 3 above)
Comment (V-Gira) [reply src/script/components/Modals/GroupCreation/GroupCreationModal.tsx]: and maybe add the import for
```
import {StatusCodes as HTTP_STATUS} from 'http-status-codes';
```
and
```suggestion
        const NON_FEDERATING_BACKENDS = HTTP_STATUS.CONFLICT.;
```

**Review by V-Gira — APPROVED** [review]

**Pull request merged**